app.js

```diff
@@ -0,0 +1,46 @@
+const express = require("express");
+const app = express();
+const path = require("path");
+const fs = require("fs/promises");
+const { getTopics } = require("./db/controllers/topics.controller");
+const { getArticleById } = require("./db/controllers/articles.controller");
+
+app.use(express.json());
+
+// Route to get topics
+app.get("/api/topics", getTopics);
+
+// Route to get an article by ID
+// app.get("/api/articles/:article_id", getArticleById);
+
+// Route to get API documentation
+app.get("/api", (req, res, next) => {
+  const filePath = path.join(__dirname, "endpoints.json");
+
+  fs.readFile(filePath, "utf-8")
+    .then((data) => {
+      const endpoints = JSON.parse(data);
+      res.status(200).json(endpoints);
```

Comment:
All this logic could be packed away into a separate controller file. We can just require the endpoints.json file in, then send it directly from a controller